### DISCUSSION TITLE
Add Hyprlock and hypridle

### DISCUSSION THREAD

**Discussion by mahaveergurjar** [prasanthrangan/hyprdots#1047] [Ideas]
Add hyprlock and hypridle i think  these are more stable than swaylock and swayidle

**Comment by prasanthrangan** [comment]
yup, `Hyprlock` and `hypridle` is already in my list... but haven't tried it out yet

**Comment by mahaveergurjar** [comment]
OK

**Comment by dieBakterie** [comment]
What's the current state for hyprlock's and hypridle's implementation?

**Comment by mahaveergurjar** [comment]
the hyprlock and hyprland are work in main branch but in hyde branch need to modified the script if you used main branch hyprdots then got to this link [https://github.com/kRHYME7/Hyprdesk](url) downloads hypridle and hyprlock files

**Reply by dieBakterie** [reply]
OK thanks for the info.

**Reply by dieBakterie** [reply]
And when i installed using hyde?

**Reply by mahaveergurjar** [reply]
the the script that is used in hyplock that put wallbash and wallpaper that are current wallpaper set this are not set in hyprlock because the wallpaper are move from swww to .local/share/bin only these are not work all hyprlock work

**Reply by dieBakterie** [reply]
If i want to use hyprlock/hypridle i'll need to use the Hyprdesk repo? Btw the Link is invalid

**Reply by kRHYME7** [reply]
> If i want to use hyprlock/hypridle i'll need to use the Hyprdesk repo? Btw the Link is invalid

Hey don't use that repo its' outdated now  if you like I can add  draft for hyprlock where you can copy the files. is that cool?

**Reply by dieBakterie** [reply]
Yeah it is and sorry for the late reply

**Comment by se7en-x230** [comment]
@kRHYME7 how do you get the album art on hyprlock? I use your setup and script but for whatever reason the mpris.art is created in .cache/hypdots/ but not shown

**Reply by kRHYME7** [reply]
Oww I'm just testing it. It's useless for now because the album won't be updated after lock. I'm waiting a for PR to be merged. Also I've seen that we can't really reproduce yet the font size. We need another Feature from hyrlock to make it easy. manipulate those values. let's wait for another month I guess haha.

**Reply by kRHYME7** [reply]
BTW you should have YT or spotify playing in the background for the album to function.

**Reply by se7en-x230** [reply]
I play my music form my cli file manager via mpv and it shows the the now playing song BUT I have not checked if it updates once the song changes.

**Reply by se7en-x230** [reply]
Ok just checked. Tracks update in my case on the lockscreen

**Reply by kRHYME7** [reply]
> Ok just checked. Tracks update in my case on the lockscreen

It will update only if you trigger the lockscreen
heres   the MR https://github.com/hyprwm/hyprlock/pull/247

**Reply by kRHYME7** [reply]
https://github.com/hyprwm/hyprlock/issues/225#event-12384590620

now it's merge Yey

**Reply by se7en-x230** [reply]
SEEEEXXY... 

![image](https://github.com/prasanthrangan/hyprdots/assets/69777713/767684a8-3aae-4558-9d27-85a96caed3e4)

**Reply by kRHYME7** [reply]
source the mpris.conf  widget
```
$artFILE = /tmp/hyde-mpris.png

$fn_splash=echo "$(hlock_mpris.sh || hyprctl splash)"

# SPLASH
label {
    monitor = 
    text = cmd[update:1000] $fn_splash
    color = $text_1_rgba
    font_family = JetBrainsMono, Font Awesome 6 Free Solid
    position = 0, -50
    halign = center
    valign = bottom
}


image {
    monitor = 
    path = $artFILE
    size = 400 # lesser side if not 1:1 ratio
    rounding = 1 # negative values mean circle
    border_size = 4
    border_color = rgb(221, 221, 221) # $primary_3_rgba
    rotate = 0 # degrees, counter-clockwise
    reload_time = 0 # seconds between reloading, 0 to reload with SIGUSR2
    reload_cmd = # command to get new path. if empty, old path will be used. don't run "follow" commands like tail -F

    position = 0, 200
    halign = center
    valign = bottom
}

```

then the script to generate the thumbs  I call it ``` hlock_mpris.sh ``` 
for me, I added it to PATH
```
#! /bin/env bash

THUMB=/tmp/hyde-mpris
# THUMB=$HOME/.cache/hyde

fetch_thumb() {
artUrl=$(playerctl metadata --format '{{mpris:artUrl}}') 
[[ "${artUrl}" = "$(cat "${THUMB}.inf")" ]]  && roman_clock.sh -c && return 0

printf "%s\n" "$artUrl" > "${THUMB}.inf"

curl -so "${THUMB}.png" "$artUrl"
convert "${THUMB}.png" -quality 50 "${THUMB}.png" && mv "${THUMB}.png" "${THUMB}.png"
pkill -USR2 hyprlock
}

# playerctl status | grep -q "Playing" && fetch_thumb 

{ playerctl metadata --format '{{title}}    {{artist}}' && fetch_thumb ;} || { rm -f "${THUMB}*" && exit 1;}


```

**Reply by se7en-x230** [reply]
&& roman_clock.sh does what? 

even without the roman_clock script I do not manage at the moment.

my structure is
hyprlock
├── layout1.conf
└── widgets
    ├── background.conf
    ├── date.conf
    ├── day.conf
    ├── general.conf
    ├── greetings.conf
    ├── input-field.conf
    ├── mpris.conf
    └── time.conf

2 directories, 9 files

modified the mpris.conf have the hlock_mpris.sh in my .local/bin/

**Reply by se7en-x230** [reply]
~cf ❯ hlock_mpris.sh
I Put A Spell On You (Floorplan Remix)    Nina Simone

It creates the thumbnail as it looks in a 2 sec interval in /temp/ but I still have trouble to show it on the hyprlock

**Reply by kRHYME7** [reply]
is your .local/bin in $PATH ? 

> && roman_clock.sh does what?

you can omit that it just convert my clock into roman_numberals


try to run ``` hyprlock -c path/to/mpris.conf```

**Reply by kRHYME7** [reply]
ow update your hyprlock to git ``` yay -Sy hyprlock-git ```

**Reply by se7en-x230** [reply]
yes sure ~/.local/bin/ is in the $PATH

but it looks like the convert part does not work correctly
[LOG] Framebuffer created, status 36053
mv: '/tmp/hyde-mpris.png' and '/tmp/hyde-mpris.png' are the same file
mv: '/tmp/hyde-mpris.png' and '/tmp/hyde-mpris.png' are the same file
mv: '/tmp/hyde-mpris.png' and '/tmp/hyde-mpris.png' are the same file
mv: '/tmp/hyde-mpris.png' and '/tmp/hyde-mpris.png' are the same file
[LOG] Authenticating
[LOG] auth: authenticated for hyprlock

convert "${THUMB}.png" -quality 50 "${THUMB}.png" 
&& mv "${THUMB}.png" "${THUMB}.png"              #### why move the file again?

**Reply by se7en-x230** [reply]
hyprlock-git !!!
That did it ... I thought I am crazy already :)

**Comment by se7en-x230** [comment]
![PXL_20240407_095306076](https://github.com/prasanthrangan/hyprdots/assets/69777713/c1ec24a1-c7cc-4a8d-8c21-c1545717b55d)

**Comment by dieBakterie** [comment]
Is it against #house-rules-🪦 to ask(no) so What's the actual state for the implementation of hyprlock and hypridle in HyDE?

**Reply by kRHYME7** [reply]
We can definitely share hyprlock configs but if we add them by default the repo will be flooded with issues as the configs are not scaled. Some got crazy scaling.  I will check some reference later on how can we manage the scaling. (I personally got two monitors and scaling it is hard as I got 1 high DPI) 

Planning to make a hyprlock PR though so we can share ideas there. Good with that ?

**Reply by dieBakterie** [reply]
Yeah
